src/version.ts

```diff
@@ -1,6 +1,41 @@
 import { semver } from "./deps.ts";
 
-export class Version {
+export interface Version {
+  main: string;
+  app?: string;
+  prerelease: boolean;
+
+  compare(other: Version): number;
+  satisfies(spec: string): boolean;
+}
+
+const COMMIT_REGEX = /^[0-9a-f]{40}$/;
+
+export function makeVersion(main: string, app?: string, prerelease?: boolean) {
+  if (COMMIT_REGEX.test(main)) {
+    return new CommitVersion(main);
+  } else {
+    return new SemanticVersion(main, app, prerelease);
+  }
+}
+
+export class CommitVersion implements Version {
+  constructor(public readonly main: string) {}
+
+  public get prerelease() {
+    return false;
+  }
+
+  compare(other: Version): number {
+    return this.main.localeCompare(other.main);
```

Comment:
How does this work?

Reply:
Do you mean that commit hashes shouldn't be sorted? The fetcher that returns commit versions only return a list of 1 so this shouldn't even really be called but I suppose it should just never be sorted to begin with.